specs/sharding/beacon-chain.md

```diff
@@ -673,7 +690,16 @@ def process_shard_header(state: BeaconState, signed_header: SignedShardBlobHeade
     # Verify that the header is within the processing time window
     assert header_epoch in [get_previous_epoch(state), get_current_epoch(state)]
     # Verify that the shard is active
-    assert shard < get_active_shard_count(state, header_epoch)
+    start_shard = get_start_shard(state, slot)
+    committees_per_slot = get_committee_count_per_slot(state, header_epoch)
+    end_shard = start_shard + committees_per_slot
+    shard_count = get_active_shard_count(state, header_epoch)
+    # Per slot, there may be max. shard_count committees.
+    # If there are shard_count * 2/3 per slot, then wrap around.
+    if end_shard >= shard_count:
+        assert not (end_shard - shard_count <= shard < start_shard)
+    else:
+        assert start_shard <= shard < end_shard
```

Comment:
May be it makes sense to add a helper function `is_shard_active_at_slot()` to make the logic a bit less obfuscated?
I believe it would also be handy in other places (like Gossip messages validation)

Reply:
This was cleaned up already, please review the latest set of changes.